stats/compute_test.go

```diff
@@ -0,0 +1,123 @@
+package stats_test
+
+import (
+	"errors"
+	"math/rand"
+	"reflect"
+	"testing"
+
+	"github.com/benchttp/worker/stats"
+)
+
+func TestCompute(t *testing.T) {
+	t.Run("passing invalid dataset returns error", func(t *testing.T) {
+		for _, testcase := range []struct {
+			name string
+			data []float64
+			want error
+			zero bool // whether or not Compute response may be partially written
+		}{
+			{
+				name: "empty dataset",
+				data: []float64{},
+				want: stats.ErrEmptySlice,
+				zero: true,
+			},
+			{
+				name: "not enough values",
+				data: newDataStub(2), // not enough for 9 deciles
+				want: &stats.ComputeError{},
+				zero: false,
+			},
+		} {
+			t.Run(testcase.name, func(t *testing.T) {
+				res, err := stats.Compute(testcase.data)
+
+				if err == nil {
+					t.Error("want error, got none")
+				}
+
+				if !errors.As(err, &testcase.want) {
+					t.Errorf("want %T, got %+v", testcase.want, err)
+				}
+
+				switch {
+				case testcase.zero && !reflect.ValueOf(res).IsZero():
+					t.Errorf("want stats output to be zero value, got %+v", res)
+				case !testcase.zero && reflect.ValueOf(res).IsZero():
+					t.Error("want stats output to be non-zero value, got zero value")
+				}
+			})
+		}
+	})
+
+	t.Run("happy path", func(t *testing.T) {
+		t.Run("happy path", func(t *testing.T) {
+			var (
+				size = 10000
+				want = stats.Stats{
+					Min:    1,
+					Max:    10000,
+					Mean:   5000,
+					Median: 5000,
+					// StdDev:  0,
+					Deciles: [9]float64{1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000},
+				}
+			)
+
+			data := newDataStub(size)
+			got, err := stats.Compute(data)
+			if err != nil {
+				t.Fatalf("want nil error, got %v", err)
+			}
+
+			if reflect.ValueOf(got).IsZero() {
+				t.Error("want stats output to be non-zero value, got zero value")
+			}
+
+			for _, stat := range []struct {
+				name string
+				want float64
+				got  float64
+			}{
+				{"min", want.Min, got.Min},
+				{"max", want.Max, got.Max},
+				{"mean", want.Mean, got.Mean},
+				{"median", want.Median, got.Median},
+				// {"standard deviation", want.StdDev, got.StdDev},
```

Comment:
[question] Why commented out?

Reply:
Because math is hard, and sometimes I get super strange results like `24597h25m38.648435536s` and I need to investigate a bit more.

Out of the scope of this PR. Will add a comment.